examen/oral/v1/program.asm

```diff
@@ -0,0 +1,75 @@
+extern fgets
+extern printf
+extern stdin
+extern strlen
+extern malloc
+
+section .data
+	fmt_string: db "%s", 0xd, 0xa, 0
+	fmt_decimal: db "%d", 0xd, 0xa, 0
+    len dd 128
+
+section .bss
+    arr resd 1
+
+section .text
+global main
+
+do_something:
+    push ebp
+    mov ebp, esp
+
+    mov eax, dword [ebp + 8]
+    mov ebx, dword [ebp + 12]
+    mov ecx, dword [ebp + 16]
+    xor edx, edx
+repeat:
+    mov dl, byte [ebx + ecx - 1]
+    mov byte[eax], dl
+    inc eax
+    dec ecx
+    jnz repeat
+
+    leave
+    ret
+
+main:
+    enter 28, 0
+    mov dword[ebp-0x4], 0x1234abcd
```

Comment:
I haven't seen ``enter`` instruction used that much in the labs. What do you think about using an explicit instruction to move the SP?

Reply:
E pe slide-uri.